### PR TITLE
[video_player] Added new option allowBackgroundPlayback

### PR DESCRIPTION
**Pull request by IlyaMax**
https://github.com/flutter/plugins/pull/4324 reopened this pr

Duplicated description from above issue
This PR introduces a new option to VideoPlayerOptions, which enables user to decide on whether user wants to keep playing video in the background or not.

This does NOT add functionality of video playback with controls. It could be achieved with audio_service package for instance.

Related issues
Fixes https://github.com/flutter/flutter/issues/63836
## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/video_player/video_player/test/video_player_test.dart [hunk 1]
```diff
@@ -98,6 +98,22 @@ class _FakeClosedCaptionFile extends ClosedCaptionFile {
 }
 
 void main() {
+  void verifyPlayingWhenAppLifecyclePaused(
+    VideoPlayerController controller, {
+    required bool isObserving,
+  }) {
+    final wasPlayingBeforePause = controller.value.isPlaying;
+    WidgetsBinding.instance!
```
Comment (stuartmorgan) [inline packages/video_player/video_player/test/video_player_test.dart:106]: You'll need to add [`_ambiguate`](https://github.com/flutter/plugins/blob/1df2543c78d034cc82cc9ac0f63865f8dc8bf31e/packages/video_player/video_player/lib/video_player.dart#L1042-L1047) to this file and call it on these lines to fix the analyzer issue on master.

File: packages/video_player/video_player/pubspec.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ description: Flutter plugin for displaying inline video with other Flutter
   widgets on Android, iOS, and web.
 repository: https://github.com/flutter/plugins/tree/main/packages/video_player/video_player
 issue_tracker: https://github.com/flutter/flutter/issues?q=is%3Aissue+is%3Aopen+label%3A%22p%3A+video_player%22
-version: 2.2.18
+version: 2.3.0
```
Comment (stuartmorgan) [inline packages/video_player/video_player/pubspec.yaml:7]: You need to update the video_player_platform_interface minimum version to require the version that added the new option. As written someone's project could pass through package resolution with a version that wouldn't compile.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by EconomicEgret** [comment]
Hi @IlyaMax, thanks so much for doing this work.

I really need this for a project soon so I've picked up and continued this PR in #5013.  I hope you're doing ok!

**Comment by IlyaMax** [comment]
I close this as you can continue

**Pull request closed without merging**